### PR TITLE
Open the files as binary for python3

### PR DESCRIPTION
**Pull request by mscherer**
Since the files do have unicode in it, we need to make sure
they are opened as binary on python 3

Fix issue #1301 and make code run on python3 with
non french locales.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ def load_version():
     # load all vars into globals, otherwise
     #   the later function call using global vars doesn't work.
     globals_dict = {}
-    with open(os.path.join('nilearn', 'version.py')) as fp:
+    with open(os.path.join('nilearn', 'version.py'), 'rb') as fp:
         exec(fp.read(), globals_dict)
```
Comment (aabadie) [inline setup.py]: but then you need to decode the byte stream here using `fp.read().decode()`.

File: setup.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def is_installing():
 _VERSION_GLOBALS = load_version()
 DISTNAME = 'nilearn'
 DESCRIPTION = 'Statistical learning for neuroimaging in Python'
-with open('README.rst') as fp:
+with open('README.rst', 'rb') as fp:
     LONG_DESCRIPTION = fp.read()
```
Comment (aabadie) [inline setup.py]: same here.

**Comment by mscherer** [comment]
And now, it fail on python 2

**Comment by AlexandreAbraham** [comment]
> And now, it fail on python 2

The error is : `SyntaxError: encoding declaration in Unicode string`

This is due to the fact that `version.py` starts with an encoding declaration `# *- encoding: utf-8 -*-`. No idea how to solve this though.

**Comment by mscherer** [comment]
Yup, if we keep the encoding declaration, it fail on python3, if we don't, it fail on python2.
We could remove the first line on python 3, or add it dynamically for python 2, but neither sound great.

**Comment by aabadie** [comment]
I could make it work on both python 2 and 3 by removing the encoding declaration `# *- encoding: utf-8 -*-` in `version.py` and using `io.open('README.rst/version.py', encoding='utf-8')` instead of the call to `open`.
Not sure if it's ok for you.

**Comment by mscherer** [comment]
That would work, but then, people editing the file might not have the good encoding (in theory, I suspect that in practice, that's ok). Let's try to see how it work with ci.

**Comment by mscherer** [comment]
So no, this doesn't work, it make other part of the code fail:
```
ERROR: Failure: SyntaxError (Non-ASCII character '\xc3' in file /home/travis/build/nilearn/nilearn/testvenv/local/lib/python2.7/site-packages/nilearn-0.2.6-py2.7.egg/nilearn/version.py on line 4, but no encoding declared; see http://www.python.org/peps/pep-0263.html for details (version.py, line 4))
```

**Comment by kchawla-pi** [comment]
Is this alive? If not, please close it by this weekend.

**Pull request closed without merging**